poethepoet/context.py

```diff
@@ -105,11 +106,15 @@ def get_executor(
     ) -> "PoeExecutor":
         from .executor import PoeExecutor
 
+        cwd_option = task_options.get("cwd", ".")
+        if cwd_option == '$exec_cwd':
```

Comment:
Stylistically having the magic value look like an env var is inconsistent, and could be confusing. 

So I would prefer one of either:
A) making the cwd option accept an environment variable (the same way [some other options do](https://github.com/nat-n/poethepoet/blob/75ef547e8c1802e96661fd9b5c2da8d20930a5ec/poethepoet/task/args.py#L269)), which can be passed `"${PWD}"` as a value (with some special logic to make it work the same seamlessly on windows); or
B) use a magic keyword that doesn't look like an env var like `"parent_cwd"` (I'm undecided on the best naming semantics for this because normally just "cwd" would mean what we want here :/ )

At this moment I'm not convinced any particular solution is best... more thought required. One nice thing about option A is that it could make it possible for the cwd to depend on a user defined argument.

Reply:
I don't hate option A, although we would need to ensure that the value of `PWD` is the actual startup working directory. `PWD` is normally defined by the shell, which may not always be present, I presume. Maybe we should use a different name (like `PARENT_PWD` as you suggested for option B) so that it does not confuse users, and is explicitly the type of value we want it to be, since it would be defined internally.

As for option B, I would still prefer to have some special character that is unlikely to be a part of an actual directory name (however unlikely it may be that a directory will be called `cwd`). Maybe something like `##parent_cwd##` would work?

Reply:
Thinking about this again, I'm leaning more towards option A. I agree that the issue of having to make the magic value not be path like is a bit of a design smell, but I also like the incidental power of accepting env vars :). 

As for PWD always being available from the environment... maybe the most elegant/delightful solution would be to make `${PWD}` always resolve to the result of `os.getcwd()`, whether or not it's inside a shell that works that way. What do you think?

Probably this would imply extending the `EnvManager::fill_template` function to accept an optional kwarg that could accept a mapping of override variables like `{"PWD": os.getcwd()}`. Either that or we add a new env var –maybe `$POE_INVOKE` to complement the existing `$POE_ROOT` that points to the dir with the pyproject.toml.

Reply:
Yea, thinking about it more, I'm also liking option A quite a bit - could lead to some interesting use-cases.

As for `PWD`, I'd personally still lean more towards defining a custom variable, maybe `${POE_PWD}`? This could work the following way: if it's not provided (i.e. the user did not set it beforehand), we use `os.getcwd()` to define it; if it *is* provided, we may well reuse the provided value. This way the semantics would be pretty clear when suitably documented. I feel like the shell's `$PWD` variable is kind of dubious as it is and I'd be more inclined to avoid using it if we can.

For what it's worth, I feel like extending the `EnvManager` method would be slightly overcomplicating the whole matter, more so than it needs to be. But I don't mean to impose, if you think it would be useful perhaps in more places, we may do it that way :)

Reply:
OK, lets go with setting POE_PWD as you suggest in EnvManager [along side POE_ROOT](https://github.com/nat-n/poethepoet/blob/main/poethepoet/env/manager.py#L52). As to what happens if POE_ROOT is already set: we need to think through the implications of always setting it to `os.getcwd()` (or not) in scenarios where the poe CLI is invoked by another poe task (e.g. monorepo usecase). I'm not sure it makes sense to consider that the user might explicitly set POE_PWD, but inheriting it from the parent poe task *might* make sense.

Reply:
I added POE_PWD as instructed, and covered it by tests as much as I could tell would be useful. As far as I can tell, it seems to all be working, but please do let me know if I have overlooked something :)